app/api/utils/staticFilesMiddleware.ts

```diff
@@ -1,18 +1,30 @@
-import { Request, Response } from 'express';
+import { NextFunction, Request, Response } from 'express';
 import { pathFunction, fileExists } from 'api/files';
+import { createError } from '.';
+
+const checkFilePath = async (fileName: string, filePath: string) => {
+  if (!fileName || !(await fileExists(filePath))) {
+    throw createError('file not found', 404);
+  }
+};
 
 const staticFilesMiddleware = (pathFunctions: pathFunction[]) => async (
   req: Request,
-  res: Response
+  res: Response,
+  next: NextFunction
 ) => {
   const pathToUse = await pathFunctions.reduce<Promise<pathFunction>>(async (current, path) => {
     if (await fileExists(path(req.params.fileName))) {
       return path;
     }
     return current;
   }, Promise.resolve(pathFunctions[0]));
-
-  res.sendFile(pathToUse(req.params.fileName));
+  try {
+    await checkFilePath(req.params.fileName, pathToUse(req.params.fileName));
+    res.sendFile(pathToUse(req.params.fileName));
+  } catch (e) {
+    next();
```

Comment:
Although I'm not sure if this could be used, I think it's better to pass the exception to the next function always that it's in into a catch